src/System.Net.Http/tests/FunctionalTests/HttpCookieProtocolTests.cs

```diff
@@ -151,6 +155,68 @@ public async Task GetAsync_AddCookieHeader_CookieHeaderSent()
             });
         }
 
+        [OuterLoop("Uses external server")]
+        [Theory, MemberData(nameof(EchoServers))]
+        public async Task SendAsync_RemoteServersWithCookies_Success(Uri remoteServer)
+        {
+            var expectedVersion = new Version(1,1);
+            HttpClientHandler handler = CreateHttpClientHandler();
+
+            if (remoteServer.Host == Configuration.Http.Http2Host)
+            {
+                expectedVersion = new Version(2,0);
+                TestHelper.EnsureHttp2Feature(handler);
+            }
+
+            using (HttpClient client = new HttpClient(handler))
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, remoteServer)
+                {
+                    Version = expectedVersion,
+                };
+
+                // Make remote server send SetCookie header.
+                request.Headers.Add("X-SetCookie", "name=value");
+                request.Headers.Add("X-SetCookie", "name1=value1");
+                request.Headers.Add("X-SetCookie", "name2=value2");
+
+                using (HttpResponseMessage response = await client.SendAsync(request))
+                {
+                    Assert.Equal(expectedVersion, response.Version);
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                    // Verify server sends back SetCookie header.
+                    Assert.Contains("Set-Cookie: name=value, name1=value1, name2=value2", response.Headers.ToString());
+
+                    // Server does not have any cookies yet.
+                    Assert.Contains("\"Cookies\": {}", await response.Content.ReadAsStringAsync());
+                }
+
+                // Send next request to see if the cookie has been wrote to the header.
+                var newRequest = new HttpRequestMessage(HttpMethod.Get, remoteServer)
+                {
+                    Version = expectedVersion,
```

Comment:
This will fail on NETFX tests when running against the HTTP/2 server.  You can't set HttpRequestMessage.Version = new Version(2,0).

You can either SkipOnTargetFramework the test method.  Or you can change this code:

```c#
            if (remoteServer.Host == Configuration.Http.Http2Host)
            {
                expectedVersion = new Version(2,0);
                TestHelper.EnsureHttp2Feature(handler);
            }
```
to this

```c#
            if (remoteServer.Host == Configuration.Http.Http2Host && !PlatformDetection.IsFullFramework)
            {
                expectedVersion = new Version(2,0);
                TestHelper.EnsureHttp2Feature(handler);
            }
```

Reply:
I chose to SkipOnTargetFramework(NETFX) the test method:

1. Manually added Cookie header will be ignored if sent with container cookies - this is a behavior difference with all .NET Core HttpHandlers. (even different than CurlHandler, which is the reverse - will only send manually added cookies)
2. HTTP/2 is not supported on NetFX.

To make the test work on NETFX will require many `if` checks or modify test case. Since we already understand its behavior, will just Skip on NETFX here.